src/units/abilities.cpp

```diff
@@ -495,25 +519,34 @@ bool unit::ability_active_impl(const std::string& ability,const config& cfg,cons
 	return true;
 }
 
-bool unit::ability_affects_adjacent(const std::string& ability, const config& cfg,int dir,const map_location& loc,const unit& from) const
+bool unit::ability_affects_adjacent(const std::string& ability, const config& cfg, unsigned int dist, int dir, const map_location& loc, const unit& from) const
 {
+	if(!cfg.has_child("affect_adjacent")) {
+		return false;
+	}
 	bool illuminates = ability == "illuminates";
 
 	assert(dir >=0 && dir <= 5);
-	//correct reversion of direction
-	map_location::direction direction{ dir = dir < 3 ? dir + 3 : dir - 3 };
+	map_location::direction direction{ dir };
+	unsigned int radius = 1;
 
 	for (const config &i : cfg.child_range("affect_adjacent"))
 	{
+		if(i["radius"] != "all_map") {
+			radius = i["radius"].to_int(1);
+			if(radius == 0) {
```

Comment:
Should this be `radius <= 0`?

Reply:
value is unsigned, how could be negative?

Reply:
replaced by radius <= 0